testsuite/forge/src/tx_emitter.rs

```diff
@@ -0,0 +1,717 @@
+        let balance = retrieve_account_balance(&client, faucet_account.address()).await?;
+        for b in balance {
+            if b.currency.eq(XUS_NAME) {
+                println!(
+                    "DD account current balances are {}, requested {} coins",
+                    b.amount, coins_total
+                );
+                break;
+            }
+        }
+        Ok(faucet_account)
+    }
+
+    pub async fn get_seed_accounts(
+        &mut self,
+        json_rpc_clients: &[JsonRpcClient],
+        seed_account_num: usize,
+    ) -> Result<Vec<LocalAccount>> {
+        info!("Creating and minting seeds accounts");
+        let tx_factory = self.tx_factory.clone();
+        let mut i = 0;
+        let mut seed_accounts = vec![];
+        while i < seed_account_num {
+            let mut client = self.pick_mint_client(json_rpc_clients).clone();
+            let mut creation_account = &mut self.network_info.treasury_compliance_account;
+            let mut batch = gen_random_accounts(min(MAX_TXN_BATCH_SIZE, seed_account_num - i));
+            let create_requests = batch
+                .iter()
+                .map(|account| {
+                    create_parent_vasp_request(
+                        creation_account,
+                        account.authentication_key(),
+                        tx_factory.clone(),
+                    )
+                })
+                .collect();
+            execute_and_wait_transactions(&mut client, &mut creation_account, create_requests)
+                .await?;
+            i += batch.len();
+            seed_accounts.append(&mut batch);
+        }
+        info!("Completed creating seed accounts");
+
+        Ok(seed_accounts)
+    }
+
+    pub async fn mint_accounts(
+        &mut self,
+        req: &EmitJobRequest,
+        requested_accounts: usize,
+    ) -> Result<()> {
+        if self.accounts.len() >= requested_accounts {
+            info!("Not minting accounts");
+            return Ok(());
+        }
+        let num_accounts = requested_accounts - self.accounts.len(); // Only minting extra accounts
+        let coins_per_account = SEND_AMOUNT * MAX_TXNS;
+        let coins_total = coins_per_account * num_accounts as u64;
+        let tx_factory = self.tx_factory.clone();
+        let client = self.pick_mint_client(&req.json_rpc_clients);
+
+        // Create seed accounts with which we can create actual accounts concurrently
+        let seed_accounts = self
+            .get_seed_accounts(&req.json_rpc_clients, req.json_rpc_clients.len())
+            .await?;
+        let faucet_account = self.get_money_source(coins_total).await?;
+        let actual_num_seed_accounts = seed_accounts.len();
+        let num_new_child_accounts =
+            (num_accounts + actual_num_seed_accounts - 1) / actual_num_seed_accounts;
+        let coins_per_seed_account = coins_per_account * num_new_child_accounts as u64;
```

Comment:
We can just name `actual_num_seed_accounts` to `num_seed_accounts`

This is another magical calculation for me:
`num_accounts` is the number new accounts
`num_new_child_accounts` is what exactly?  How many children do we have?  What's the significance does every account have 1 child?

Reply:
I agree there is a bit confusion, I will think some better variables name here.
Generally, we create numbers of seed accounts(VASPs) first, `actual_num_seed_accounts` stands for this.
and we do split all required accounts(child vasp) equally under each seed account(VASP), and we create those account by each seed(VASP) parallel, and mint coins by the same time from seed(VASP) to each required account(child VASP), `num_new_child_accounts` stands for number of each group under one seed(VASP) to create.